### PR TITLE
fix warnings

### PR DESCRIPTION
**Pull request by allentiak**
* FormValidationFailedException: add default SerialVersionUID

* Section3DateSelectionHandler: fix logic

### CODE REVIEW

File: src/main/java/com/yilmaznaslan/errorhandling/FormValidationFailedException.java [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package com.yilmaznaslan.errorhandling;
 
 public class FormValidationFailedException extends Exception {
-    public FormValidationFailedException(String message) {
+    private static final long serialVersionUID = 1L;
```
Comment (yilmaznaslan) [inline src/main/java/com/yilmaznaslan/errorhandling/FormValidationFailedException.java:4]: Why is it needed

File: src/main/java/com/yilmaznaslan/formhandlers/Section3DateSelectionHandler.java [hunk 2]
```diff
@@ -117,7 +117,7 @@ protected void selectTimeslot(Select select) throws InterruptedException {
         List<WebElement> availableHours = select.getOptions();
         for (int i = 0; i < availableHours.size(); i++) {
             String timeSlot = availableHours.get(i).getText();
-            if (timeSlot != null || !timeSlot.equals("")) {
+            if (timeSlot != null && !timeSlot.isBlank()) {
```
Comment (yilmaznaslan) [inline src/main/java/com/yilmaznaslan/formhandlers/Section3DateSelectionHandler.java:120]: Why change from || to && ?

**Review by yilmaznaslan — COMMENTED** [review]
Why are these changes needed I didn't understand

File: src/main/java/com/yilmaznaslan/errorhandling/FormValidationFailedException.java (hunk 1 above)
Comment (allentiak) [reply src/main/java/com/yilmaznaslan/errorhandling/FormValidationFailedException.java:4]: To suppress the following Eclipse warning:
"The serializable class FormValidationFailedException does not declare a static final serialVersionUID field of type long"

**Comment by allentiak** [comment]
Or, in other words, I'm just trying to fix some "broken windows".
https://www.artima.com/articles/dont-live-with-broken-windows

File: src/main/java/com/yilmaznaslan/formhandlers/Section3DateSelectionHandler.java (hunk 2 above)
Comment (yilmaznaslan) [reply src/main/java/com/yilmaznaslan/formhandlers/Section3DateSelectionHandler.java:120]: Now got it, thanks for improving the quality @allentiak :)

File: src/main/java/com/yilmaznaslan/errorhandling/FormValidationFailedException.java (hunk 1 above)
Comment (yilmaznaslan) [reply src/main/java/com/yilmaznaslan/errorhandling/FormValidationFailedException.java:4]: I ' suggest  that we shouldn't add functionality or complexity until it's actually needed. 

Given that we don't currently face serialization-related problems, this overhead doesn't seem to be necessary. You can ignore the warning ;). I highly suggest to you use sonarlint plugin btw instead custom eclipse code analyzer

**Review by yilmaznaslan — CHANGES_REQUESTED** [review]
Please rever the change in the Exception

File: src/main/java/com/yilmaznaslan/errorhandling/FormValidationFailedException.java (hunk 1 above)
Comment (allentiak) [reply src/main/java/com/yilmaznaslan/errorhandling/FormValidationFailedException.java:4]: I see (and agree on) your point in avoiding over-engineering.

That being said, I don't see how adding *one line* that gets rid of a compiler warning can be a case of over-engineering - specially considering that the workaround implies either using an IDE-specific plugin or adding another line of code (in this case, an annotation)...

Comment (allentiak) [reply src/main/java/com/yilmaznaslan/errorhandling/FormValidationFailedException.java:4]: My comment above addresses https://github.com/yilmaznaslan/berlin-auslaenderbehorde-termin-bot/pull/75#pullrequestreview-1677812358

**Comment by allentiak** [comment]
I will certainly fix the formatting.
If you know of an Eclipse plugin that can format the code as IntelliJ does, please do let me know.

**Comment by allentiak** [comment]
> Please rever the change in the Exception

Please see https://github.com/yilmaznaslan/berlin-auslaenderbehorde-termin-bot/pull/75#discussion_r1359261094

**Pull request closed without merging**